components/shopify/actions/search-customer/search-customer.mjs

```diff
@@ -0,0 +1,59 @@
+import shopify from "../../shopify.app.js";
+
+export default {
+  key: "shopify-search-customer",
+  name: "Search Customer",
+  description: "Search for a customer or a list of customers",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    shopify,
+    ids: {
+      type: "string[]",
+      propDefinition: [
+        shopify,
+        "customerId",
+        (c) => c,
+      ],
+      label: "Customer IDs",
+      description: `Restrict results to customers specified by a comma-separated list of IDs
+        Options will display the email registered with the ID
+        It is possible to select more than one option`,
+      optional: true,
+    },
+    fields: {
+      propDefinition: [
+        shopify,
+        "responseFields",
+      ],
+    },
```

Comment:
What's the reason to include a `fields` prop for this and other "search" actions? Shopify's REST Admin API has a [request-based rate limit](https://shopify.dev/api/usage/rate-limits), so users can fetch all fields without penalty.

Reply:
Issue addressed